contracts/staking/contracts/src/staking_pools/MixinStakingPool.sol

```diff
@@ -111,92 +110,17 @@ contract MixinStakingPool is
-    function joinStakingPoolAsMaker(bytes32 poolId)
-        external
-    {
-        // Is the maker already in a pool?
-        address makerAddress = msg.sender;
-        IStructs.MakerPoolJoinStatus memory poolJoinStatus = _poolJoinedByMakerAddress[makerAddress];
-        if (poolJoinStatus.confirmed) {
-            LibRichErrors.rrevert(LibStakingRichErrors.MakerPoolAssignmentError(
-                LibStakingRichErrors.MakerPoolAssignmentErrorCodes.MakerAddressAlreadyRegistered,
-                makerAddress,
-                poolJoinStatus.poolId
-            ));
-        }
-
-        poolJoinStatus.poolId = poolId;
-        _poolJoinedByMakerAddress[makerAddress] = poolJoinStatus;
-
-        // Maker has joined to the pool, awaiting operator confirmation
-        emit PendingAddMakerToPool(
-            poolId,
-            makerAddress
-        );
-    }
-
-    /// @dev Adds a maker to a staking pool. Note that this is only callable by the pool operator.
-    /// Note also that the maker must have previously called joinStakingPoolAsMaker.
-    /// @param poolId Unique id of pool.
-    /// @param makerAddress Address of maker.
-    function addMakerToStakingPool(
-        bytes32 poolId,
-        address makerAddress
-    )
-        external
-        onlyStakingPoolOperatorOrMaker(poolId)
-    {
-        _addMakerToStakingPool(poolId, makerAddress);
-    }
-
-    /// @dev Removes a maker from a staking pool. Note that this is only callable by the pool operator or maker.
-    /// Note also that the maker does not have to *agree* to leave the pool; this action is
-    /// at the sole discretion of the pool operator.
-    /// @param poolId Unique id of pool.
-    /// @param makerAddress Address of maker.
-    function removeMakerFromStakingPool(
-        bytes32 poolId,
-        address makerAddress
-    )
-        external
-        onlyStakingPoolOperatorOrMaker(poolId)
-    {
-        bytes32 makerPoolId = getStakingPoolIdOfMaker(makerAddress);
-        if (makerPoolId != poolId) {
-            LibRichErrors.rrevert(LibStakingRichErrors.MakerPoolAssignmentError(
-                LibStakingRichErrors.MakerPoolAssignmentErrorCodes.MakerAddressNotRegistered,
-                makerAddress,
-                makerPoolId
-            ));
-        }
-
-        // remove the pool and confirmation from the maker status
-        delete _poolJoinedByMakerAddress[makerAddress];
-        _poolById[poolId].numberOfMakers = uint256(_poolById[poolId].numberOfMakers).safeSub(1).downcastToUint32();
-
-        // Maker has been removed from the pool`
-        emit MakerRemovedFromStakingPool(
-            poolId,
-            makerAddress
-        );
-    }
-
-    /// @dev Returns the pool id of the input maker.
-    /// @param makerAddress Address of maker
-    /// @return Pool id, nil if maker is not yet assigned to a pool.
-    function getStakingPoolIdOfMaker(address makerAddress)
+    function setMakerStakingPool(bytes32 poolId)
```

Comment:
I kinda like keeping this names `joinPoolAsMaker`, I think it's a bit more descriptive. Otherwise it's not obvious that the caller should be the maker.